src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1401,3 +1429,98 @@ def _reorder_cache(past_key_values, beam_idx):
         for layer_past in past_key_values:
             reordered_past += (tuple(past_state.index_select(0, beam_idx) for past_state in layer_past),)
         return reordered_past
+
+
+@add_start_docstrings(
+    """
+    Whisper Encoder Model with a sequence classification head on top (a linear layer over the pooled output) for tasks
+    like SUPERB Keyword Spotting.
+    """,
+    WHISPER_ENCODER_INPUTS_DOCSTRING,
+)
+class WhisperForAudioClassification(WhisperPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.encoder = WhisperEncoder(config)
+        num_layers = config.num_hidden_layers + 1  # transformer layers + input embeddings
+        if config.use_weighted_layer_sum:
+            self.layer_weights = nn.Parameter(torch.ones(num_layers) / num_layers)
```

Comment:
We should probably be careful about the dtype
```suggestion
            self.layer_weights = nn.Parameter(torch.ones(num_layers, dtype = self.dtype) / num_layers)
```

Reply:
I don't think this is required? For instance we don't pass `dtype` to any of our linear layers.

Reply:
If it is it would require a full change to our library, e.g. in T5:
https://github.com/huggingface/transformers/blob/f7ca656f07537762d03d9f4084fbcfe7246095d0/src/transformers/models/t5/modeling_t5.py#L244

Reply:
Indeed sorry !

Reply:
(Was confused with thw attention dtype 😅 )